make/sandbox-ui.mk

```diff
@@ -0,0 +1,162 @@
+deploy-sandbox-ui: HOST_OPERATOR_API=https://$(shell oc get route api -n ${HOST_NS} -o custom-columns=":spec.host" | tr -d '\n')
+deploy-sandbox-ui: RHDH=https://rhdh-${SANDBOX_UI_NS}.$(shell oc get ingress.config.openshift.io/cluster -o jsonpath='{.spec.domain}')
+deploy-sandbox-ui:
+	$(MAKE) check-sso-credentials
+	@echo "sandbox ui will be deployed in '${SANDBOX_UI_NS}' namespace"
+	$(MAKE) create-namespace SANDBOX_UI_NS=${SANDBOX_UI_NS}
+ifeq ($(PUSH_SANDBOX_IMAGE),true)
+	$(MAKE) push-sandbox-plugin
+endif
+	kustomize build deploy/sandbox-ui/ui-e2e-tests | REGISTRATION_SERVICE_API=${REGISTRATION_SERVICE_API} \
+			HOST_NS=${HOST_NS} \
+			HOST_OPERATOR_API=${HOST_OPERATOR_API} \
+			SANDBOX_UI_NS=${SANDBOX_UI_NS} \
+			SANDBOX_PLUGIN_IMAGE=${IMAGE_TO_PUSH_IN_QUAY} \
+			RHDH=${RHDH} envsubst | oc apply -f -
+	$(MAKE) configure-oauth-idp
+	@echo "restarting registration-service to apply toolchainconfig changes"
+	@oc -n ${HOST_NS} rollout restart deploy/registration-service
+	@oc -n ${SANDBOX_UI_NS} rollout status deploy/rhdh
+	@echo "Developer Sandbox UI running at ${RHDH}"
+
+
+check-sso-credentials:
+	@echo "checking SSO credentials..."
+	@if [ -z "$$SSO_USERNAME" ] || [ -z "$$SSO_PASSWORD" ]; then \
+		echo "SSO_USERNAME or SSO_PASSWORD not set"; \
+		exit 1; \
+	fi
+	@status=$$(curl -s -o /dev/null -w "%{http_code}" \
+	  -X POST "https://sso.devsandbox.dev/auth/realms/sandbox-dev/protocol/openid-connect/token" \
+	  -d "grant_type=password" \
+	  -d "client_id=sandbox-public" \
+	  -d "username=$$SSO_USERNAME" \
+	  -d "password=$$SSO_PASSWORD"); \
+	if [ "$$status" != "200" ]; then \
+	  echo "failed trying to login to 'https://sso.devsandbox.dev/auth/realms/sandbox-dev' ($$status) — check your SSO credentials."; \
+	  exit 1; \
+	fi
+
+configure-oauth-idp:
+	@echo "configuring DevSandbox identity provider"
+	@oc create secret generic ${OPENID_SECRET_NAME} \
+		--from-literal=clientSecret=dummy \
+		--namespace=openshift-config
+	OPENID_SECRET_NAME=${OPENID_SECRET_NAME} envsubst < deploy/sandbox-ui/ui-e2e-tests/oauth-idp-patch.yaml | \
+		oc patch oauths.config.openshift.io/cluster --type=merge --patch-file=/dev/stdin
+
+create-namespace:
+	@if ! oc get project ${SANDBOX_UI_NS} >/dev/null 2>&1; then \
+		echo "Creating namespace ${SANDBOX_UI_NS}"; \
+		oc new-project ${SANDBOX_UI_NS} >/dev/null 2>&1 || true; \
+	else \
+		echo "Namespace ${SANDBOX_UI_NS} already exists"; \
+	fi
+	@oc project ${SANDBOX_UI_NS} >/dev/null 2>&1
+
+
+.PHONY: get-rhdh-plugins
+get-rhdh-plugins:
+ifeq ($(strip $(RHDH_PLUGINS_DIR)), $(TMPDIR)rhdh-plugins)
+	echo "using rhdh-plugins repo from master"
+	@$(MAKE) clone-rhdh-plugins
+else
+	echo "using local rhdh-plugins repo: ${RHDH_PLUGINS_DIR}"
+endif
+
+.PHONY: clone-rhdh-plugins
+clone-rhdh-plugins:
+	rm -rf ${RHDH_PLUGINS_DIR}; \
+	git clone https://github.com/redhat-developer/rhdh-plugins $(RHDH_PLUGINS_DIR) && \
```

Comment:
btw, the rhdh-plugins seems to be a bit bigger repository, but you could do only the shallow copy to speed it a bit up.